plugins/modules/win_firewall.ps1

```diff
@@ -33,6 +37,8 @@ Try {
     ForEach ($profile in $firewall_profiles) {
 
         $currentstate = (Get-NetFirewallProfile -Name $profile).Enabled
+        $current_inboundaction = (Get-NetFirewallProfile -Name $profile).DefaultInboundAction
+        $current_outboundaction = (Get-NetFirewallProfile -Name $profile).DefaultOutboundAction
```

Comment:
You are calling `Get-NetFirewallProfile` 3 times here, call it once and set each var from there

```suggestion
        $current_profile = Get-NetFirewallProfile -Name $profile
        $currentstate = $current_profile.Enabled
        $current_inboundaction = $current_profile.DefaultInboundAction
        $current_outboundaction = $current_profile.DefaultOutboundAction
```